src/bin/cargo/commands/fix.rs

```diff
@@ -79,7 +77,12 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
     let lockfile_path = args.lockfile_path(gctx)?;
     ws.set_requested_lockfile_path(lockfile_path.clone());
 
-    let mut opts = args.compile_options(gctx, mode, Some(&ws), ProfileChecking::LegacyTestOnly)?;
+    let mut opts = args.compile_options(
+        gctx,
+        CompileMode::Check { test },
```

Comment:
I feel like it's clearer by inlining this

Reply:
This is not a doc change. I would expect it being in a separate PR.